tdrs-backend/tdpservice/users/test/test_auth.py

```diff
@@ -153,486 +150,326 @@ def test_login_fails_without_state(api_client):
-    }
-    mock_post.return_value = MockRequest(data=token)
-    mock_decode.return_value = decoded_token
-    factory = APIRequestFactory()
-    view = TokenAuthorizationOIDC.as_view()
-    request = factory.get("/v1/login", {"state": state, "code": code})
-    request.session = api_client.session
-    request.session["state_nonce_tracker"] = {
-        "nonce": nonce,
-        "state": state,
-        "added_on": time.time(),
+@pytest.fixture()
+def states_factory():
+    """Bundle together nonce, state, and code for tests."""
+    yield {
+        'nonce':   "testnonce",
+        'state':    "teststate",
+        'code':     secrets.token_hex(32)
     }
-    response = view(request)
-    assert response.status_code == status.HTTP_302_FOUND
-
 
-@pytest.mark.django_db
-def test_login_with_old_email(
-    api_client,
-    mock_token,
-    mocker,
-    patch_login_gov_jwt_key,
-    user
-):
-    """Login should work with existing user."""
-    user.username = "test_old_email@example.com"
-    user.save()
-    nonce = "testnonce"
-    state = "teststate"
-    code = secrets.token_hex(32)
-    mock_post = mocker.patch("tdpservice.users.api.login.requests.post")
-    token = {
-        "access_token": "hhJES3wcgjI55jzjBvZpNQ",
-        "token_type": "Bearer",
-        "expires_in": 3600,
-        "id_token": mock_token,
-    }
-    mock_decode = mocker.patch("tdpservice.users.api.login.jwt.decode")
-    decoded_token = {
-        "email": "test_new_email@example.com",
-        "email_verified": True,
-        "nonce": nonce,
-        "iss": "https://idp.int.identitysandbox.gov",
-        "sub": user.login_gov_uuid,
-        "verified_at": 1577854800,
-    }
-    mock_post.return_value = MockRequest(data=token)
-    mock_decode.return_value = decoded_token
+@pytest.fixture()
+def req_factory(states_factory, mock, api_client):
+    """Generate a client request for API usage, part of DRY."""
+    states = states_factory
     factory = APIRequestFactory()
-    view = TokenAuthorizationOIDC.as_view()
-    request = factory.get("/v1/login", {"state": state, "code": code})
+    request = factory.get(
+            "/v1/login",
+            {
+                "state": states['state'],
+                "code": states['code']
+            }
+            )
     request.session = api_client.session
-    request.session["state_nonce_tracker"] = {
-        "nonce": nonce,
-        "state": state,
-        "added_on": time.time(),
-    }
-    response = view(request)
-    # Ensure the user's username was updated with new email.
-    assert User.objects.filter(username="test_new_email@example.com").exists()
-    assert response.status_code == status.HTTP_302_FOUND
+    yield request
 
-
-@pytest.mark.django_db
-def test_login_with_initial_superuser(
-    api_client,
-    mock_token,
-    mocker,
-    patch_login_gov_jwt_key,
-    settings,
-    user
+def decoded_token(
+    email,
+    nonce,
+    sub="b2d2d115-1d7e-4579-b9d6-f8e84f4f56ca",
```

Comment:
```suggestion
    sub="uuid.uuidv4()",
```
This *might* work too but I'm not 100% sure.